backend/routes/content.py

```
@@ -559,3 +565,66 @@ def get_bonus_challenges():
+@content_bp.route("/hints/<int:hint_id>/buy", methods=["POST"])
+@jwt_required()
+def buy_hint(hint_id):
+    try:
+        user_id = get_jwt_identity()
+        user_hint = UserHint.query.filter_by(user_id=user_id, hint_id=hint_id).first()
+        if user_hint is None:
+            user_hint = UserHint(user_id=user_id, hint_id=hint_id)
+        # deduct gems for hint purchase
+        user = User.query.get(user_id)
+        if user.gems < GEMS_FOR_HINT:
+            logger.error(f"Insufficient gems to buy hint for user {user_id}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)